### DISCUSSION TITLE
Use Azure Verified Modules

### DISCUSSION THREAD

**Discussion by raix** [platformplatform/PlatformPlatform#640] [Ideas]
### Detailed description?

**Is your feature request related to a problem? Please describe.**
We are currently creating custom bicep files for setting up infrastructure but Microsoft provides "secure by default" bicep modules which is also used internally between teams in Microsoft to avoid teams creating custom bicep modules.

**Describe the solution you'd like**
As much as possible replace custom bicep modules with AVMs - A PoC might be a good first step to check:
* if any changes improve security or not
* access is given via service principal vs connection strings/key vault (might result in being a larger task)

**Describe alternatives you've considered**
Currently running a mix of the current modules and AVMs - but it would be great with consistency and potential upstream updates / improvements

**Additional context**
[Azure-Verified-Modules](https://azure.github.io/Azure-Verified-Modules/indexes/bicep/)

### Feature Type

Infrastructure

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by tjementum** [comment]
I've had this on my list for over a year: https://github.com/orgs/platformplatform/projects/2/views/1?filterQuery=azure+ver&pane=issue&itemId=52860004

But if one uses Azure Verified Modules, the `--what-if` in Bicep stops working. You simply lose all insights into what changes are going to be applied to your Azure resources. Because of this issue: https://github.com/Azure/arm-template-whatif/issues/157. 

I like to only have things on my roadmap that I'm committed to doing, so I will likely archive this issue.
